src/main/webapp/script.js

```diff
@@ -88,3 +90,45 @@ function getJsonIfOk(response) {
   }
 }
 
+/**
+ * Handles response by checking it and converting it to text.
+ */
+function handleTextResponse(response) {
```

Comment:
I also had a servlet that returned a simple text message, and because of that the original postData() js function could not be used. However, don't you think that we could just always return JSON instead of text, and then we could keep the JSON parsing inside the postData() function? (There is a simple Java JSONObject that you can use instead of an own class https://stleary.github.io/JSON-java/org/json/JSONObject.html.) But maybe an APPLICATION_JSON request takes more time, then I also support the text response

Reply:
Yes, I will change the servlet to return a JSON (I opened the issue #24 for that).
I returned the text only to make it in time for the demo.

Reply:
I'm ok if you change this after this PR since it works the same way for the user.
This is an example of taking on technical debt to be refactored away later in order to be faster. It's always worth considering if this is worth it and necessary or if the refactoring should instead happen before any of the code is merged into master.